### DISCUSSION TITLE
v4.1.3 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#4851] [Releases]
## Highlights
* **QEMU backend updated to v7.2.0**. Also updated usbredir (0.13.0), virglrenderer (latest commit), ANGLE (latest commit)
* **Improved renderer backend**. Lots of bug fixes and stability improvements to GPU accelerated Linux VMs (most common crashes when GPU acceleration is enabled should be gone now). New option to change the renderer backend to ANGLE Metal (which is also the default for iOS now) and limit FPS.
* (iOS) **TrollStore build (UTM.HV.ipa) now supports JIT**. Additionally, there is now in-app settings.
* (macOS) **AppleScript (OSA) support and CLI interface**. You can control parts of UTM through the OSA interface. Currently there is support for listing VMs as well as start/stop/suspend operations and the ability to print out the guest serial port connections. More functionality will be added in the future. A command line application is also provided in `UTM.app/Contents/MacOS/utmctl` (which you can symlink to `/usr/local/bin/utmctl` if desired) that can be used for automation tasks without needing to learn AppleScript.
* (macOS 13+) **Menu bar extra**. You can enable the menu bar extra icon in Preferences (Cmd+,) as well as disable the dock icon. The menu bar extra provides a minimal interface to start/stop/suspend VMs and is useful when paired with headless VMs (a VM that does not have any display or terminal console installed).

## Notes
* **iOS 14 and macOS 11.3** are the minimum supported systems for UTM v4.x. Please use UTM v3.x for support down to iOS 11 and macOS 11
* The configuration backend has been massively rewritten. **Please backup all VMs prior to updating as you will not be able to re-open VMs saved by UTM v4 on older versions of UTM if you decide to downgrade.**

## Changes (v4.1.3)
* Updated QEMU to v7.2.0
* Reworked CocoaSpice renderer to respond faster to GL updates from the guest, should improve frame pacing on GL supported Linux VMs
* Added global setting to change renderer backend, introduce ANGLE Metal as an option (and also the new default for iOS)
* Added global setting to limit FPS which can be used to improve frame pacing for a smoother rendering experience
* Settings: clearly show if the current display hardware supports GPU acceleration
* Wizard: GPU acceleration is now enabled by default when Linux is selected as the operating system
* Fixed crash due to an error with `GL_DRAW_COOKIE_INVALID` (#4722)
* Fixed deadlock on startup caused by gstosxaudio (#2364)
* Fixed memory leak when GPU acceleration is enabled (#4449)
* Fixed crash in `-[MTLResourceList releaseAllObjectsAndReset]` (#3519)
* Fixed Mesa issue causing some GNOME desktop to freeze when GPU acceleration is enabled
* Fixed incorrect logic when cloning a VM (#4815)
* Fixed some thread scheduling warning (#4352)
* Localization: updated French (thanks @FRtranslator)
* (iOS) TrollStore build: use `com.apple.private.security.no-sandbox` instead of `com.apple.private.security.no-container` on recommendation of other devs and to align with how other apps perform JIT.
* (iOS) Fixed blank screen when choosing "Open" in the wizard (#4842)
* (iOS) Fixed toolbar not auto-hiding before first interaction (#4844)
* (macOS) New design for global settings (Cmd+,) which better organizes the settings
* (macOS) utmctl: Fixed crash when `utmctl help attach` runs (#4817)
* (macOS) utmctl: Fixed `utmctl stop` with `--request` and `kill` (#4850)

## Changes (v4.1.2)
* Localization: updated Spanish (Latin America) (thanks @JezerM)
* Localization: updated Japanese (thanks @MMP0 and @skyarrow87)
* Localization: updated Chinese Simplified (thanks @changanmoon)
* (iOS) Fix new VM wizard not appearing on < iOS 14.5 (#4776)
* (macOS) Implement OSA interface for AppleScript
* (macOS) Introduce a CLI application (`UTM.app/Contents/MacOS/utmctl`) that wraps around the OSA interface for automation
* (macOS 13+) New settings option to enable a menu bar extra icon for controlling VMs
* (macOS 13+) New settings option to disable the dock icon
* (macOS) Fix various alignment issues in settings (#4596) (thanks @JezerM)
* (macOS) Fix a crash when the pasteboard is updated (#4745)
* (macOS) Support custom resolutions for Apple Virtualization VMs
* (macOS) Fixed a UI freeze when canceling out of the settings for an Apple Virtualization VM

## Changes (v4.1.1)
* Fixed launching SPARC machines (#4575)
* Fixed an error when attempting to save a VM where one of its removable drives is linked to a file that has been deleted
* Fixed IPSW/.utm download failing when the device goes to sleep (the download should resume after wake up)
* Restored scroll bars when system is set to always show scroll bars. Added a new SwiftUI crash workaround thanks to @evelyneee.
* Display all sizes in binary (MiB/GiB) mode in order to be consistent (#4396)
* Localization: added Polish (thanks @mavethee)
* Localization: updated Japanese (thanks @MMP0)
* Localization: updated Spanish (Latin America) (thanks @JezerM)
* Localization: updated French (thanks @FRtranslator)
* (macOS) Fixed the port forward UI on macOS 11 and introduce a new port forward UI for macOS 12+ (#4690)
* (macOS) Fixed "This virtual machine cannot be run on this machine." error on a newly created Apple VM (#4595)
* (macOS) Added new 21:9 widescreen resolution for Apple VMs (thanks @Tandolf)
* (macOS) Added a clarifying message about shared directory support on older macOS guests (#4594)
* (macOS) Added new shutdown options (software request and force kill) which can be toggled by long pressing the power button in the VM window (#4001)
* (macOS) Added a confirmation dialog when deleting a drive (#4687)

## Changes (v4.1.0)
* Updated QEMU to v7.1.0
* Updated ANGLE and virglrenderer to latest commit
* Fixed an issue where user override of machine property settings was not applied properly
* Localization: updated Japanese (thanks @MMP0)
* Localization: updated Spanish (Latin America) (thanks @JezerM)
* (iOS) Added in-app settings popover which synchronizes with the Settings app (useful for TrollStore installs where the Settings app page is inaccessible) (#4446)
* (iOS) TrollStore: support JIT without needing a separate JIT enabler (thanks @PojavLauncherTeam for the idea)

## Issues
Please check the full list on [Github][2] and help report any bug you find that is not listed.

* **UTM SE is currently not working!** (#4848)

## Installation

* [iOS](https://docs.getutm.app/installation/ios/)
* [macOS](https://docs.getutm.app/installation/macos/)

| File | Description | Installation | JIT | Hypervisor | USB |
|------|------------|--------------|-----|-----------|-----|
| UTM.deb | Jailbroken iOS version | Open in Cydia, dpkg, or Sileo | Yes | Yes(1) | Yes |
| UTM.dmg | macOS version | Mounting and copying UTM.app to /Applications | Yes | Yes | Yes |
| UTM.ipa | Non-jailbroken iOS version (sideloading) | AltStore, etc (see guide) | Yes(2) | No | No |
| UTM.HV.ipa | Non-jailbroken iOS version (TrollStore) | TrollStore | Yes | Yes(1) | Yes |
| UTM.SE.ipa | Non-jailbroken iOS version (sideloading) | AltStore, enterprise signing, etc | No | No | No |

1. Hypervisor on iOS requires an M1 iPad.
2. Enabling JIT may require a separate JIT enabler such as [Jitterbug][3] or Jitstreamer.

[1]: https://getutm.app/install/
[2]: https://github.com/utmapp/UTM/issues
[3]: https://github.com/osy/Jitterbug

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v4.1.3'>v4.1.3 (Beta)</a>.</em>

**Comment by Intecpsp** [comment]
I see the wizard defaults to `virtio-gpu-gl-pci` now for Linux VMs. I have legacy Linux VMs targeting `virtio-ramfb`. Is it recommended to flip those to `virtio-gpu-gl-pci` or `virtio-ramfb-gl`?

**Reply by osy** [reply]
Sure and help test the new graphics engine. But it’s not required if you want to keep everything as is.

**Reply by Intecpsp** [reply]
I tried `virtio-gpu-gl-pci` and it was drawing on the screen super slowly. I saw parts of the boot sequence I had never seen because of how fast it was with `virtio-ramfb`. I flipped to `virtio-ramfb-gl` and it's back to being super fast.

**Reply by osy** [reply]
That’s an issue with the EFI drivers. ramfb-gl just gives EFI a different surface to work with but once the OS is booted they’re exactly the same.
